src/mlpack/core/math/ccov.hpp

```diff
@@ -0,0 +1,255 @@
+/**
+ * @file ccov.hpp
+ * @author Ryan Curtin
+ * @author Conrad Sanderson
+ *
+ * ccov(X) is same as cov(trans(X)) but without the cost of computing trans(X)
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_MATH_CCOV_HPP
+#define MLPACK_CORE_MATH_CCOV_HPP
+
+namespace mlpack {
+namespace math /** Miscellaneous math routines. */ {
+
+template<typename eT>
+inline
+arma::Mat<eT>
+ccov(const arma::Mat<eT>& A, const arma::uword norm_type = 0)
```

Comment:
Since we are making this a part of the mlpack codebase now, can we name this `ColumnCovariance()` instead of `ccov()` for consistency?  Also we can switch `uword` to `size_t` (again for consistency).

Reply:
@rcurtin , Done